File: tests/pages/cart.page.ts
import { Page, expect } from "@playwright/test";
import data from '../data/data.json';
import { productSlug } from '../support/utils';

export class CartPage {
  constructor(private page: Page) {};

  async open() {
    await this.page.goto(data.cartPage);
  }

  inventoryItem() { return this.page.locator('[data-test="inventory-item"]') }
  itemName() { return this.page.locator('[data-test="inventory-item-name"]') }
  removeProduct(name: string) { return this.page.locator(`[data-test="remove-${name}"]`) }

  async removeItem(name: string) {
    const slug = productSlug(name);
    const removeButton = this.removeProduct(slug);
    await expect(removeButton).toBeVisible();
    await removeButton.click();
  }
}

File: tests/pages/cart.page.cs
// CartPage.cs
using System.Threading.Tasks;
using Microsoft.Playwright;
using static Microsoft.Playwright.Assertions;
using MyTests.Support;

namespace MyTests.Pages;

public sealed class CartPage
{
    private readonly IPage _page;
    private readonly string _cartPageUrl;

    public CartPage(IPage page, string cartPageUrl)
    {
        _page = page;
        _cartPageUrl = cartPageUrl;
    }

    // NAVIGATION
    public Task OpenAsync() => _page.GotoAsync(_cartPageUrl);

    // LOCATORS
    public ILocator InventoryItem => _page.Locator("[data-test=\"inventory-item\"]");
    public ILocator ItemName => _page.Locator("[data-test=\"inventory-item-name\"]");
    public ILocator RemoveProduct(string name) => _page.Locator($"[data-test=\"remove-{name}\"]");
    public ILocator CheckoutButton => _page.Locator("[data-test=\"checkout\"]");

    // ACTIONS
    public async Task RemoveItemAsync(string name)
    {
        var slug = Utils.ProductSlug(name); // your productSlug equivalent
        var removeButton = RemoveProduct(slug);
        await Expect(removeButton).ToBeVisibleAsync();
        await removeButton.ClickAsync();
    }

    public async Task ClickCheckoutAsync()
    {
        await Expect(CheckoutButton).ToBeVisibleAsync();
        await CheckoutButton.ClickAsync();
    }
}
